src/main/java/com/karankumar/bookproject/backend/service/AuthorService.java

```diff
@@ -50,17 +51,19 @@ public Author findById(Long id) {
     @Override
     public void save(Author author) {
         if (author != null) {
-            List<Long> matchingAuthors = findAll().stream()
+            List<Author> existingAuthors = findAll();
+            List<Long> matchingAuthors = existingAuthors.stream()
                 .map(BaseEntity::getId)
                 .filter(id -> id.equals(author.getId())).collect(Collectors.toList());
-            if (matchingAuthors.size() == 1) {
+            // Null checking will do since it will return 1 or null or empty result since ids are unique
+            if (!CollectionUtils.isEmpty(matchingAuthors)) {
                 // if an author with the same ID exists, set this incoming author ID to null so that
                 // a new row in the table is made rather than updating the row that has the same ID
                 LOGGER.log(Level.INFO, "Matching authorIds: " + matchingAuthors);
-                author.removeId();
+                authorRepository.save(new Author(author.getFirstName(), author.getLastName()));
```

Comment:
I am not sure about this logic here, all we are trying to find out if the id of the author to be saved already exists in the DB? If that's the case, shouldn't we just call some method from repo like findById and pass the id of the author to be saved?

Reply:
Actually thinking a bit more about it, seems like in any case we always want to add a new row so shouldn't we just save it as a new author always without even checking anything?

Reply:
i can change the query to findById since we are just validating if the author existing. shall I proceed with the changes?